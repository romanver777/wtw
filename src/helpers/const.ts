export const genres: Array<string> = [
  "Все жанры",
  "Комедии",
  "Документальное",
  "Драма",
  "Ужасы",
  "Семейное",
  "Романтические",
  "Фантастика",
  "Триллеры",
];

export const tabsName: Array<string> = ["Описание", "Информация", "Отзывы"];

export const similarMovies = "Вам может понравиться";

export const page: { [key: string]: string } = {
  movieCard: "movie-card",
  userPage: "user-page",
};

export const userPageTitle = {
  signIn: "Выполните вход",
  myList: "Мой список",
};
