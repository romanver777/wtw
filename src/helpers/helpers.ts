export const convertRatingLevel = (rt: string | null): string => {
  const rating = Number(rt);
  if (rating > 0 && rating < 4) return "Плохо";
  if (rating >= 4 && rating < 5) return "Удовлетворительно";
  if (rating >= 5 && rating < 6) return "Хорошо";
  if (rating >= 6 && rating < 7) return "Очень хорошо";
  if (rating >= 7 && rating < 8) return "Отлично";
  if (rating >= 8) return "Превосходно";
  return "-";
};

export const convertDateTime = (date: string): string => {
  return new Date(date).toLocaleString("ru-RU", {
    year: "numeric",
    month: "long",
    day: "numeric",
    hour: "numeric",
    minute: "numeric",
  });
};
