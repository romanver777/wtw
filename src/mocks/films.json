{
  "films": [
    {
      "filmId": 1302121,
      "nameRu": "Соединённые Штаты против Билли Холидей",
      "nameEn": "The United States vs. Billie Holiday",
      "year": "2021",
      "filmLength": "2:10",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "биография"
        },
        {
          "genre": "музыка"
        }
      ],
      "rating": "5.8",
      "ratingVoteCount": 586,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1302121.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1302121.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1171895,
      "nameRu": "Неуязвимый (сериал)",
      "nameEn": "Invincible",
      "year": "2021-...",
      "filmLength": null,
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "мультфильм"
        },
        {
          "genre": "ужасы"
        },
        {
          "genre": "фантастика"
        }
      ],
      "rating": "8.1",
      "ratingVoteCount": 1549,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1171895.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1171895.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1245486,
      "nameRu": "Невероятные (сериал)",
      "nameEn": "The Nevers",
      "year": "2021-...",
      "filmLength": null,
      "countries": [
        {
          "country": "Великобритания"
        },
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "фантастика"
        },
        {
          "genre": "фэнтези"
        },
        {
          "genre": "боевик"
        }
      ],
      "rating": "7.0",
      "ratingVoteCount": 882,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1245486.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1245486.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1015471,
      "nameRu": "Райя и последний дракон",
      "nameEn": "Raya and the Last Dragon",
      "year": "2021",
      "filmLength": "1:47",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "мультфильм"
        },
        {
          "genre": "фэнтези"
        },
        {
          "genre": "боевик"
        }
      ],
      "rating": "7.6",
      "ratingVoteCount": 43386,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1015471.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1015471.jpg",
      "ratingChange": null
    },
    {
      "filmId": 571335,
      "nameRu": "Бесстыжие (сериал)",
      "nameEn": "Shameless",
      "year": "2011-2021",
      "filmLength": "0:55",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "комедия"
        }
      ],
      "rating": "8.6",
      "ratingVoteCount": 207954,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/571335.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/571335.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1328036,
      "nameRu": "Happy End (сериал)",
      "nameEn": null,
      "year": "2021-...",
      "filmLength": "0:35",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "комедия"
        }
      ],
      "rating": "7.4",
      "ratingVoteCount": 3282,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1328036.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1328036.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1045583,
      "nameRu": "Конёк-Горбунок",
      "nameEn": null,
      "year": "2021",
      "filmLength": "1:51",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "фэнтези"
        },
        {
          "genre": "комедия"
        },
        {
          "genre": "приключения"
        }
      ],
      "rating": "6.8",
      "ratingVoteCount": 100901,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1045583.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1045583.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1274271,
      "nameRu": "Родные",
      "nameEn": null,
      "year": "2020",
      "filmLength": "1:40",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "комедия"
        },
        {
          "genre": "драма"
        }
      ],
      "rating": "6.4",
      "ratingVoteCount": 20539,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1274271.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1274271.jpg",
      "ratingChange": null
    },
    {
      "filmId": 767379,
      "nameRu": "Фарго (сериал)",
      "nameEn": "Fargo",
      "year": "2014-...",
      "filmLength": "0:53",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "драма"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "8.4",
      "ratingVoteCount": 119492,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/767379.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/767379.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1245524,
      "nameRu": "Зависнуть в Палм-Спрингс",
      "nameEn": "Palm Springs",
      "year": "2020",
      "filmLength": "1:30",
      "countries": [
        {
          "country": "США"
        },
        {
          "country": "Гонконг"
        }
      ],
      "genres": [
        {
          "genre": "комедия"
        },
        {
          "genre": "мелодрама"
        },
        {
          "genre": "фантастика"
        }
      ],
      "rating": "7.2",
      "ratingVoteCount": 130542,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1245524.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1245524.jpg",
      "ratingChange": null
    },
    {
      "filmId": 4430018,
      "nameRu": "Абсолютное доказательство",
      "nameEn": "Absolute Proof",
      "year": "2021",
      "filmLength": "2:00",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "документальный"
        }
      ],
      "rating": null,
      "ratingVoteCount": null,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/4430018.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/4430018.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1437860,
      "nameRu": "Чикатило (сериал)",
      "nameEn": null,
      "year": "2020-2020",
      "filmLength": "0:50",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "детектив"
        },
        {
          "genre": "драма"
        }
      ],
      "rating": "5.9",
      "ratingVoteCount": 7820,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1437860.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1437860.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1005008,
      "nameRu": "Майор Гром",
      "nameEn": null,
      "year": "2017",
      "filmLength": "0:25",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "короткометражка"
        },
        {
          "genre": "боевик"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "7.0",
      "ratingVoteCount": 29251,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1005008.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1005008.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1291310,
      "nameRu": "В её глазах (мини-сериал)",
      "nameEn": "Behind Her Eyes",
      "year": "2021-2021",
      "filmLength": "0:50",
      "countries": [
        {
          "country": "Великобритания"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "драма"
        },
        {
          "genre": "детектив"
        }
      ],
      "rating": "7.3",
      "ratingVoteCount": 12949,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1291310.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1291310.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1038789,
      "nameRu": "Охотник на монстров",
      "nameEn": "Monster Hunter",
      "year": "2020",
      "filmLength": "1:43",
      "countries": [
        {
          "country": "США"
        },
        {
          "country": "Китай"
        },
        {
          "country": "Германия"
        }
      ],
      "genres": [
        {
          "genre": "фэнтези"
        },
        {
          "genre": "боевик"
        },
        {
          "genre": "приключения"
        }
      ],
      "rating": "5.6",
      "ratingVoteCount": 24933,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1038789.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1038789.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1248786,
      "nameRu": "Аферистка",
      "nameEn": "I Care a Lot",
      "year": "2020",
      "filmLength": "1:58",
      "countries": [
        {
          "country": "США"
        },
        {
          "country": "Великобритания"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "комедия"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "6.6",
      "ratingVoteCount": 81569,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1248786.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1248786.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1193013,
      "nameRu": "Грань времени",
      "nameEn": "Synchronic",
      "year": "2019",
      "filmLength": "1:42",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "фантастика"
        },
        {
          "genre": "драма"
        }
      ],
      "rating": "6.2",
      "ratingVoteCount": 10361,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1193013.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1193013.jpg",
      "ratingChange": null
    },
    {
      "filmId": 958500,
      "nameRu": "Охотник за разумом (сериал)",
      "nameEn": "Mindhunter",
      "year": "2017-2019",
      "filmLength": "1:00",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "драма"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "8.1",
      "ratingVoteCount": 50186,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/958500.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/958500.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1347194,
      "nameRu": "Звездный рубеж",
      "nameEn": "Cosmic Sin",
      "year": "2021",
      "filmLength": "1:28",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "фантастика"
        },
        {
          "genre": "боевик"
        }
      ],
      "rating": "3.4",
      "ratingVoteCount": 524,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1347194.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1347194.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1236082,
      "nameRu": "Ассасин: Битва миров",
      "nameEn": "Ci sha xiao shuo jia",
      "year": "2021",
      "filmLength": "2:10",
      "countries": [
        {
          "country": "Китай"
        }
      ],
      "genres": [
        {
          "genre": "фэнтези"
        },
        {
          "genre": "боевик"
        }
      ],
      "rating": "6.0",
      "ratingVoteCount": 645,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1236082.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1236082.jpg",
      "ratingChange": null
    }
  ]
}
