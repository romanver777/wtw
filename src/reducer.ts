import movies from "./mocks/films.json";
import { FilmsType } from "./types/types";
import { GENRES } from "./helpers/const";

interface InitStateType {
  films: FilmsType;
  genre: string;
}

interface ActionType {
  SET_GENRE: string;
  GET_FILMS_BY_GENRE: string;
}

const InitialState: InitStateType = {
  films: [...movies],
  genre: GENRES[0],
};

const ActionType = {
  SET_GENRE: `SET_GENRE`,
  GET_FILMS_BY_GENRE: `GET_FILMS_BY_GENRE`,
};

const ActionCreator = {
  setGenre: (genre: string) => ({
    type: ActionType.SET_GENRE,
    payload: genre,
  }),
  getFIlmsByGenre: (genre, films) => ({
    type: ActionCreator.GET_FILMS_BY_GENRE,
    payload: [],
  }),
};

const reducer = (state = InitialState, action): InitStateType => {
  switch (action.type) {
    case ActionType.SET_GENRE:
      return {
        ...state,
        genre: action.payload,
      };
    default:
      return state;
  }
};

const Operation = {};

export { reducer, ActionCreator, Operation };
