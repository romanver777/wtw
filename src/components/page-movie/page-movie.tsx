import React, { useEffect } from "react";
import { RouteComponentProps } from "react-router";
import { connect } from "react-redux";

import { MovieType, ReviewType, StaffType } from "../../types/types";
import { PAGE_NAME } from "../../helpers/const";
import { getMovieById } from "../../helpers/helpers";
import MovieCardFull from "../movie-card-full/movie-card-full";
import Catalog from "../catalog/catalog";
import Footer from "../footer/footer";
import { Operation, ActionCreator } from "../../reducer";

interface StateProps {
  films: MovieType[];
  currentFilm: MovieType;
  reviews: ReviewType[];
  staff: StaffType[];
}

interface OwnProps {
  id: string;
}

interface MapProps {
  film: MovieType;
  prevFilm: MovieType;
  films: MovieType[];
  reviews: ReviewType[];
  staff: StaffType[];
}

interface DispatchProps {
  loadFilmData: (film: MovieType) => void;
}

type Props = MapProps & DispatchProps;

const PageMovie = (props: Props): JSX.Element => {
  const { film, prevFilm, loadFilmData, reviews, staff } = props;

  useEffect(() => {
    if (film !== prevFilm) loadFilmData(film);
  });

  return (
    <React.Fragment>
      {reviews && staff && (
        <React.Fragment>
          <MovieCardFull film={film} reviews={reviews} staff={staff} />
          <div className="page-content">
            <Catalog pageName={PAGE_NAME[1]} />
            <Footer />
          </div>
        </React.Fragment>
      )}
    </React.Fragment>
  );
};

const mapStateToProps = (
  state: StateProps,
  ownProps: RouteComponentProps<OwnProps>
) => {
  const { id } = ownProps.match.params;
  return {
    id,
    film: getMovieById(state.films, id) as MovieType,
    prevFilm: state.currentFilm,
    films: state.films,
    reviews: state.reviews,
    staff: state.staff,
  };
};

const mapDispatchToProps = (dispatch) => ({
  loadFilmData(film: MovieType) {
    dispatch(ActionCreator.setCurrentFilm(film));
    dispatch(Operation.loadReviews(film.filmId));
    dispatch(Operation.loadStaff(film.filmId));
  },
});

export { PageMovie };
export default connect(mapStateToProps, mapDispatchToProps)(PageMovie);
