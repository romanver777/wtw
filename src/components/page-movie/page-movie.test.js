/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from "react";
import TestRenderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { reducer } from "../../reducer";
import { Router } from "react-router";
import history from "../../history";

import { PageMovie } from "./page-movie";
import films from "../../mocks/films.json";
import { ALL_GENRES } from "../../helpers/const";

const initState = {
  films,
  awaitFilm: null,
  currentFilm: films[0],
  hoveredFilm: null,
  activeGenre: ALL_GENRES,
  genresList: null,
  tk: null,
  reviews: null,
  staff: null,
  hoveredVideoData: null,
  isMovieLoading: true,
  isMovieError: false,
};

const store = createStore(reducer, initState);

describe("PageMovie component", () => {
  const props = {
    film: films[0],
    prevFilm: null,
    loadFilmData: () => ({}),
    reviews: films[0].reviews,
    staff: films[0].staff,
    isMovieLoading: false,
    isMovieError: false,
  };
  const Trenderer = TestRenderer.create(
    <Provider store={store}>
      <Router history={history}>
        <PageMovie {...props} />
      </Router>
    </Provider>
  );
  it("renders correctly", () => {
    expect(Trenderer.toJSON()).toMatchSnapshot();
  });

  describe("renders correctly with props", () => {
    const renderer = new ShallowRenderer();

    it("loading", () => {
      const propsLoading = {
        ...props,
        isMovieLoading: true,
      };
      const res = "Loading...";
      renderer.render(<PageMovie {...propsLoading} />);
      const result = renderer.getRenderOutput();

      expect(result.props.children).toBe(res);
    });
    it("loading error", () => {
      const propsLoadingError = {
        ...props,
        isMovieLoading: true,
        isMovieError: true,
      };
      const res = "Что-то пошло не так.";
      renderer.render(<PageMovie {...propsLoadingError} />);
      const result = renderer.getRenderOutput();

      expect(result.props.text).toBe(res);
    });
    it("no film", () => {
      const propsNoFilm = {
        ...props,
        film: null,
      };
      const res = {};
      renderer.render(<PageMovie {...propsNoFilm} />);
      const result = renderer.getRenderOutput();

      expect(result.props).toStrictEqual(res);
    });
    it("no reviews", () => {
      const propsNoReviews = {
        ...props,
        reviews: null,
      };
      const res = "Loading...";
      renderer.render(<PageMovie {...propsNoReviews} />);
      const result = renderer.getRenderOutput();

      expect(result.props.children).toBe(res);
    });
    it("no staff", () => {
      const propsNoReviews = {
        ...props,
        staff: null,
      };
      const res = "Loading...";
      renderer.render(<PageMovie {...propsNoReviews} />);
      const result = renderer.getRenderOutput();

      expect(result.props.children).toBe(res);
    });
  });
});
