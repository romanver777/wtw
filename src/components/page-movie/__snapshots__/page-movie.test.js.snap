// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMovie component renders correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Соединённые Штаты против Билли Холидей"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1302121.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link "
            href="#/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            href="#/login"
            onClick={[Function]}
          >
            Вход
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Соединённые Штаты против Билли Холидей
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              драма
            </span>
            <span
              className="movie-card__genre"
            >
              биография
            </span>
            <span
              className="movie-card__genre"
            >
              музыка
            </span>
            <span
              className="movie-card__year"
            >
              2021
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <div
              className="movie-card__buttons"
            >
              <button
                className="btn btn--play movie-card__button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  height="20"
                  viewBox="0 0 19 20"
                  width="19"
                >
                  <use
                    xlinkHref="#play-s"
                  />
                </svg>
                <span>
                  Смотреть
                </span>
              </button>
              <button
                className="btn btn--list movie-card__button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  height="20"
                  viewBox="0 0 19 20"
                  width="19"
                >
                  <use
                    xlinkHref="#add"
                  />
                </svg>
                <span>
                  Сохранить
                </span>
              </button>
              <button
                className="btn  movie-card__button"
                onClick={[Function]}
                type="button"
              >
                
                <span>
                  Написать отзыв
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Соединённые Штаты против Билли Холидей"
            height="327"
            src="https://kinopoiskapiunofficial.tech/images/posters/kp/1302121.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Описание
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Информация
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Отзывы
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              5.8
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Хорошо
              </span>
              <span
                className="movie-rating__count"
              >
                586
                 голосов
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              В 1940-х правительство США начинает войну против наркотиков. Ведомые предубеждениями против цветного населения чиновники выбирают своей целью популярную певицу Билли Холидей.
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Режиссер: 
                Ли Дэниелс
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                В ролях: 
                Андра Дэй, 
                Гаррет Хедлунд, 
                Наташа Лионн, 
                Тайлер Джеймс Уильямс, 
                Треванте Роудс и другие
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <h2
        className="catalog__title"
      >
        Похожие фильмы
      </h2>
      <div
        className="catalog__movies-list"
      >
        <p>
          Не найдено
        </p>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="#/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2021 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
